src/connection-manager/agreement.ts

```diff
@@ -195,6 +194,22 @@ class Agreement {
         last_ledger_time: new Date(),
       }
 
+      let server_version
+
+      if (isPreceedingFlagLedger(validation.ledger_index)) {
+        server_version = decodeServerVersion(validation.server_version)
+        const ballot: Ballot = {
+          master_key: validation.master_key,
+          signing_key,
+          ledger_index: Number(validation.ledger_index),
+          amendments: validation.amendments?.toString(),
```

Comment:
This should be a comma-separated list or something, for easier parsing

Reply:
Yes it is comma-separated

Reply:
Really? Usually `list.toString()` generates something more like `[1, 2, 3]` instead of `1,2,3`

Reply:
`27CD95EE8E1E5A537FF2F89B6CEB7C622E78E9374EBD7DCBEDFAE21CD6F16E0A,56B241D7A43D40354D02A9DC4C8DF5C7A1F930D92A9035C4E12291B3CA3E1C2B` this is one example from db, but sure I can use `join` to make sure it's always correct

Reply:
Yeah, I'd prefer a join.
Will there be any issues with string lengths here?

Reply:
Yea that's the part I'm uncertain about, I set the length to be 10,000 for now, but not sure if there'll ever be a case when it get overflow

Reply:
That seems pretty reasonable for now, that takes 156 amendments actively being voted on which is unlikely